File: java/0122-best-time-to-buy-and-sell-stock-ii.java
class Solution {
    public int maxProfit(int[] prices) {
        int oldStockPrice = prices[0];
        int profit = 0;
        for(int i = 1; i<prices.length; i++){
            if(prices[i]<oldStockPrice){
                oldStockPrice = prices[i];
            }else{
                profit+=prices[i]-oldStockPrice;
                oldStockPrice = prices[i];
            }
        }
        return profit;
    }
}

File: kotlin/0122-best-time-to-buy-and-sell-stock-ii.kt
class Solution {
    fun maxProfit(prices: IntArray): Int {
        var profit = 0
        for(i in 0 until prices.size-1){
            if(prices[i+1] > prices[i]) profit += (prices[i+1] - prices[i])
        }
        return profit
    }
}
